Directory.Build.props

```diff
@@ -15,32 +16,134 @@
+    <MicrosoftExtensionsOptionsVersion>1.1.2</MicrosoftExtensionsOptionsVersion>
+    <MicrosoftExtensionsOptionsConfigurationExtensionsVersion>1.1.2</MicrosoftExtensionsOptionsConfigurationExtensionsVersion>
+    <MicrosoftExtensionsDependencyInjection>1.1.1</MicrosoftExtensionsDependencyInjection>
+
+    <MicrosoftDataSQLiteVersion>1.1.0</MicrosoftDataSQLiteVersion>
+    <MicrosoftPowerShell5ReferenceAssembliesVersion>1.0.0</MicrosoftPowerShell5ReferenceAssembliesVersion>
+    <MicrosoftServiceFabricServicesVersion>2.4.145</MicrosoftServiceFabricServicesVersion>
+
+    <FSharpCoreVersion>4.2.1</FSharpCoreVersion>
+
+    <SystemManagementAutomationdllVersion>10.0.10586</SystemManagementAutomationdllVersion>
+
+    <!-- 3rd party packages -->
+    <MySqlDataVersion>6.9.9</MySqlDataVersion>
+    <NewtonsoftJsonVersion>10.0.3</NewtonsoftJsonVersion>
+    <NewRelicAgentApiVersion>5.10.59.0</NewRelicAgentApiVersion>
+    <NpgsqlVersion>3.1.9</NpgsqlVersion>
+    <NSubstituteVersion>1.10.0</NSubstituteVersion>
+    <GoogleCloudPubSubV1Version>1.0.0-beta13</GoogleCloudPubSubV1Version>
+
+    <!-- Test related packages -->
+    <BenchmarkDotNetVersion>0.10.8</BenchmarkDotNetVersion>
+    <FluentAssertionsVersion>4.19.3</FluentAssertionsVersion>
+    <XunitSkippableFactVersion>1.3.3</XunitSkippableFactVersion>
+    <xUnitVersion>2.2.0</xUnitVersion>
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(BuildFlavor)' == 'Current' ">
+    <!-- System packages -->
+    <SystemNetHttpVersion>4.3.1</SystemNetHttpVersion>
+
+    <!-- Microsoft packages -->
+    <MicrosoftApplicationInsightsVersion>2.2.0</MicrosoftApplicationInsightsVersion>
+    <MicrosoftAzureEventHubsVersion>1.0.0</MicrosoftAzureEventHubsVersion>
+    <WindowsAzureStorageVersion>8.1.1</WindowsAzureStorageVersion>
+
+    <!-- 3rd party packages -->
+    <AWSSDKDynamoDBv2Version>3.3.3</AWSSDKDynamoDBv2Version>
+    <AWSSDKSQSVersion>3.3.1.7</AWSSDKSQSVersion>
+    <BondCoreCSharpVersion>5.2.0</BondCoreCSharpVersion>
+    <ConsulVersion>0.7.2.1</ConsulVersion>
+    <GoogleProtobufVersion>3.2.0</GoogleProtobufVersion>
+    <ZooKeeperNetExVersion>3.4.9.2</ZooKeeperNetExVersion>
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(BuildFlavor)' == 'Legacy' ">
+    <!-- System packages -->
+    <SystemReflectionMetadataVersion>1.4.2</SystemReflectionMetadataVersion>
+
+    <!-- Microsoft packages -->
+    <MicrosoftApplicationInsightsVersion>1.2.3</MicrosoftApplicationInsightsVersion>
+    <WindowsAzureServiceBusVersion>4.1.0</WindowsAzureServiceBusVersion>
+    <WindowsAzureStorageVersion>7.0.0</WindowsAzureStorageVersion>
+
+    <!-- 3rd party packages -->
+    <AWSSDKDynamoDBv2Version>3.1.5.2</AWSSDKDynamoDBv2Version>
+    <AWSSDKSQSVersion>3.1.0.9</AWSSDKSQSVersion>
+    <BondCoreCSharpVersion>4.0.1</BondCoreCSharpVersion>
+    <ConsulVersion>0.7.0.3</ConsulVersion>
+    <GoogleProtobufVersion>3.0.0</GoogleProtobufVersion>
+    <ZooKeeperNetExVersion>3.4.8.5</ZooKeeperNetExVersion>
```

Comment:
IMO, these packages that are probably just used in a single project, I would just move the version there and not define a global msbuild variable.